src/foam/mlang/mlang.js

```diff
@@ -2323,11 +2323,49 @@ foam.CLASS({
     }
   ],
 
+  properties: [
+    {
+      class: 'Boolean',
+      name: '_checkingNestedFObject',
+      value: false,
+      transient: true,
+      visibility: 'HIDDEN',
+      documentation: 'Support keyword search on the first level nested FObject.'
+    }
+  ],
+
   methods: [
     {
       name: 'f',
       code: function f(obj) {
-        return this.fInner_(obj, true);
+        var arg = this.arg1.f(obj);
+        if ( ! arg || typeof arg !== 'string' ) return false;
+
+        arg = arg.toLowerCase();
+
+        try {
+          var s = '';
+          const props = obj.cls_.getAxiomsByClass(foam.core.Property);
+          for ( let i = 0; i < props.length; i++ ) {
+            const prop = props[i];
+            if ( this.FObjectProperty.isInstance(prop) )
+              if ( this.checkNestedFObject(prop.f(obj)) ) return true;
+            else if ( this.Enum.isInstance(prop) )
```

Comment:
Is this else and else for the line above or the line above that? Add braces to clarify.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/3796/commits/ab66fa4be1959bd719e050f38aa0c4fc8cc9d43e.